### PR TITLE
Adding support to expand env values

### PR DESCRIPTION
**Pull request by gullitmiranda**
ex:

```sh
APP_TEST_VAR = HELLO

EXPAND=${APP_TEST_VAR} 
# ==
EXPAND=HELLO

SKIP_EXPAND=TEST_\$APP_TEST_VAR 
# == 
SKIP_EXPAND=TEST_\$APP_TEST_VAR
```

### CODE REVIEW

**Comment by stevepm** [comment]
Would be great to get this merged in!

File: lib/dotenv.ex [hunk 1]
```diff
@@ -43,6 +43,21 @@ defmodule Dotenv do
     \z
     /x
 
+  # https://regex101.com/r/XrvCwE/1
+  @env_expand_pattern ~r/
+    (?:^|[^\\])                           # prevent to expand \\$
+    (                                     # get variable key parttern
```
Comment (jarednorman) [inline lib/dotenv.ex]: This looks reasonable. Do you mind fixing this typo, then I'll merge?

**Comment by jarednorman** [comment]
I'm going to go ahead and merge this anyway as this PR is pretty stale, but should still be useful to people.

File: lib/dotenv.ex (hunk 1 above)
Comment (gullitmiranda) [reply lib/dotenv.ex]: sorry for the delay. if you are not to make direct to the master, I can make another PR with the fix

Comment (jarednorman) [reply lib/dotenv.ex]: I'll merge the fix if you PR it, but it's no big deal if you don't have time.

Comment (gullitmiranda) [reply lib/dotenv.ex]: done https://github.com/avdi/dotenv_elixir/pull/33

**Pull request merged**